src/checkers/firefoxchecker.py

```diff
@@ -0,0 +1,181 @@
+
+        if not browser_data:
+            log.warning('Invalid x-checker-data for module %s, unable to find browser source with filename %s',
+                        module_data.name,
+                        checker_data['browser']['source-filename'])
+            return None
+
+        added = []
+        processed_data = []
+        for data in external_data:
+            if data.filename in latest_info:
+                new_version = latest_info[data.filename]
+                data.state = ExternalData.State.VALID
+                if not data.current_version.matches(new_version):
+                    data.new_version = new_version
+            else:
+                if re.match(checker_data['browser']['filename-match'], data.filename) or \
+                   re.match(checker_data['translations']['filename-match'], data.filename):
+                    data.state = ExternalData.State.REMOVED
+            processed_data.append(data.filename)
+
+        added = []
+        new_sources = []
+        for source_filename, info in latest_info.items():
+            if not source_filename in processed_data:
+                source = {
+                    'filename': source_filename,
+                    'type': 'extra-data',
+                    'url': info.url,
+                    'sha256': info.checksum,
+                    'size': info.size,
+                }
+                # add new data to the same manifest as the main browser tarball
+                data = ExternalDataSource.from_source(browser_data.source_path, source, browser_data.source_parent)
+                data.state = ExternalData.State.ADDED
+                added.append(data)
+
+        return added
+
+    def _get_latest_available_version(self):
+        if self._latest_version:
+            return
+
+        latest_version_data = {}
+        request_url = self.FIREFOX_RELEASE_INFO_URL
+        with urllib.request.urlopen(request_url) as resp:
+            latest_version_data = json.load(resp)
+
+        if isinstance(latest_version_data, dict) and 'LATEST_FIREFOX_VERSION' in latest_version_data:
+            self._latest_version = latest_version_data['LATEST_FIREFOX_VERSION']
+        else:
+            self._latest_version = None
+
+    def _get_latest_version_info(self, version, checker_data):
+        results = {}
+
+        base_url = self.FIREFOX_ARCHIVE_BASE_URL.format(version=version)
+
+        url = '{}SHA256SUMS'.format(base_url)
+        with urllib.request.urlopen(url) as response:
+            sha256_table = response.read().decode()
+
+        for line in sha256_table.splitlines():
+            sha256, path = line.split(None, 1)
+            path = path.strip()
+
+            if path.startswith(self.FIREFOX_PLATFORM):
+                assert path.count('/') == 2, path
+                _, dirname, filename = path.split('/')
+
+                source_filename = None
+                if re.match(checker_data['browser']['filename-match'], filename) and \
```

Comment:
Why do we need to specify regular expressions to match in the manifest? This checker is Firefox-specific. We can just hardcode them.

Reply:
Ok, tbh I prefer to have it configurable such that we don't have to force sources/translations filename to match (e.g. in a future official firefox flatpak). 

Maybe leave the checker options but use default values if they are not set?

Reply:
A future official Firefox Flatpak will be generated as part of the Mozilla release process so will not need this script.

Reply:
> Maybe leave the checker options but use default values if they are not set?

OK! I predict that we'll never need to set them but I am prepared to be wrong :-)

Reply:
Done, ended up hardcoding the expressions in the checker - if in the future we find the need we can make it configurable again.